pkgs/tools/archivers/undmg/setup-hook.sh

```diff
@@ -1,5 +1,25 @@
 unpackCmdHooks+=(_tryUnpackDmg)
 _tryUnpackDmg() {
     if ! [[ "$curSrc" =~ \.dmg$ ]]; then return 1; fi
-    undmg "$curSrc"
+    dmgFs="$(/usr/bin/hdiutil imageinfo "$curSrc" | sed -n '/Name: disk image/s/.*(Apple_\([^ ]*FS\).*)/\1/p')"
```

Comment:
We should have a fallback to the old behavior if the program can't be run or fails.

Reply:
Sounds good. I need to think about how I'd approach this.

Reply:
The following case statement always falls back to `undmg` now. Do you find this acceptable?

Reply:
With the latest changes to this PR this is no longer relevant.